BDArmory/GameModes/ModuleSpaceFriction.cs

```diff
@@ -168,21 +171,41 @@ public void FixedUpdate()
                             vesselAlt = VAI.minAltitude;
 
                         float accelMult = 1f;
-                        if (vessel.verticalSpeed > 1) //vessel ascending
+                        if (vessel.radarAltitude < vesselAlt) 
                         {
-                            accelMult = Mathf.Clamp(Mathf.Abs((float)vessel.verticalSpeed), 1f, 100);
-                        }
-                        if (vessel.radarAltitude < Mathf.Max((vesselAlt / 10), 5))
-                        {
-                            accelMult = Mathf.Clamp((float)vessel.radarAltitude / vesselAlt, 0.3f, 1);
-                        }
-                        if (vessel.radarAltitude < vesselAlt)
-                        {
-                            for (int i = 0; i < part.vessel.Parts.Count; i++)
+                            if (vessel.radarAltitude < Mathf.Max((vesselAlt / 10), 5))
+                            {
+                                if ((vessel.situation != Vessel.Situations.LANDED && vessel.situation != Vessel.Situations.SPLASHED && vessel.situation != Vessel.Situations.PRELAUNCH) && Time.time - 5 > landedTime)
+                                {
+                                    accelMult = Mathf.Clamp(Mathf.Abs((float)vessel.verticalSpeed), 1f, 100);
+                                }
+                                else
+                                {
+                                    accelMult = 10;
+                                    landedTime = Time.time;
+                                }
+                            }
+
+                            if (VesselModuleRegistry.GetRepulsorModules(vessel).Count > 0)
+                            {
+                                using (var craftPart = VesselModuleRegistry.GetRepulsorModules(vessel).GetEnumerator())
+                                    while (craftPart.MoveNext())
+                                    {
+                                        if (craftPart.Current is null) continue;
+                                        if (craftPart.Current.part.PhysicsSignificance != 1) //attempting to apply rigidbody force to non-significant parts will NRE
+                                        {
+                                            craftPart.Current.part.Rigidbody.AddForce(((-FlightGlobals.getGeeForceAtPosition(craftPart.Current.part.transform.position) * ((part.vessel.GetTotalMass() * 10) / repulsors)) * ((vesselAlt / Mathf.Max(BodyUtils.GetRadarAltitudeAtPos(craftPart.Current.part.transform.position), 1))) / accelMult), ForceMode.Acceleration);
+                                        }
```

Comment:
Repulsor forces now divided between set parts instead of CoM; using landing gear parts seemed singularly apropos for this task, saves needing a second module/per-vessel controller

Reply:
Sounds good, but what happens if your landing gear isn't spaced nicely? Will it throw the vessel into a spin?
This sounds like it'd be a fun extra design challenge.

Reply:
Sounds highly entertaining! :-D